src/kafka_submitter.rs

```diff
@@ -115,7 +118,8 @@ pub fn share_submitted(state: &KafkaSubmitterState, user_id: &Vec<u8>, user_tag_
 				version: header.version,
 				nbits: header.bits,
 				time: header.time,
-				nonce: header.nonce,
+				hash: String::new(),  // We only include hash for weak block
```

Comment:
Hmm, can you add a Share and WeakBlockMessage, then? Seems really awkward to put an empty string in the DB, no?

Reply:
Actually the share message will normally be just processed and won't go to database directly, but will be archived as files after processing. .

Reply:
That sounds like more reason to have two separate messages? If they're handled differently why should they be the same message into kafka, or am I missing some part of how kafka works?

Reply:
Currently shares and weak blocks are treated in a same way on share message consumer side, except for checking if there's a potential valid block.